test/functional/p2p_leak_tx.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
```

Comment:
I've now reviewed this test. Sadly, it doesn't test the new code (and does not fail on master).

Since the node-under-test has only one peer, the transaction won't get added to `mapRelay` until we send the `INV` to that peer. That means in the if/else branches at the bottom of this test:

- if the node has already sent us an INV for the tx, then it'll be in the `mapRelay` and the node will send us the TX.
- if the node has not yet sent us an INV for the tx, then it won't be in the `mapRelay` and we'll fail `mi != mapRelay.end()` and not even reach `&& mi->second.m_node_set.count(pfrom->GetId())`

To test the new code, we want the to be in `mapRelay`, but the peer to not be in the `RelayEntry`. That's not possible if the node has only one peer.

Perhaps we should remove the test from this PR and add it under a future PR. If you do so, please tag me in that PR and I'll commit to reviewing the test.